### ISSUE TITLE
SES has stopped working in my app

### ISSUE THREAD

**Issue by minimalistcode** [aws/aws-sdk-ios#227]
I have used latest version of the AWS iOS SDK since it has come out (years?) to send emails with SES. I am working on an update to an app and I realized that SES no longer works. Here's what I do know:

- The version of my app that I released on 28 May 2015 can still send emails fine using SES.
- I can still go to the AWS Console and send test SES emails using the same addresses as the apps uses. So SES seems to be working fine.
- I am working on an update to my app and I just realized that SES emails I send with the same code in the app are no longer getting sent.
- I do not get any errors back from AWS. Everything looks fine.The emails just don't get sent.
- I compared my code between the 28 May 2015 release and what I have now, and the only difference I see is that I have a newer version of the AWS iOS SDK (the latest), and I converted from BTask to AWSTask.  I did this throughout my app which uses SimpleDB, SES, SNS, Cognito, and S3, and everything else works fine except SES.

I'll paste my code below, but other than changing from BTask to AWSTask I see no difference. Has something changed in SES?  Are there any suggestions on how to debug this since I'm not getting any errors back from AWS?

```
-(void) sendEmailFrom:(NSString *)fromEmailAddress to:(NSString *)toEmailAddress subject:(NSString *)subject body:(NSString *)body completionHandler:(void (^)(NSError *error))completionHandler
{
    AWSSESSendEmailRequest *awsSESSendEmailRequest = [AWSSESSendEmailRequest new];
    awsSESSendEmailRequest.source = fromEmailAddress;
    AWSSESDestination *awsSESDestination = [AWSSESDestination new];
    awsSESDestination.toAddresses = [NSMutableArray arrayWithObjects:toEmailAddress,nil];
    awsSESSendEmailRequest.destination = awsSESDestination;
    
    AWSSESMessage *awsSESMessage = [AWSSESMessage new];
    AWSSESContent *awsSESSubject = [AWSSESContent new];
    awsSESSubject.data = subject;
    awsSESSubject.charset = @"UTF-8";
    awsSESMessage.subject = awsSESSubject;
    
    AWSSESContent *awsSESContent = [AWSSESContent new];
    awsSESContent.data = body;
    awsSESContent.charset = @"UTF-8";
    
    AWSSESBody *awsSESBody = [AWSSESBody new];
    awsSESBody.text = awsSESContent;
    awsSESMessage.body = awsSESBody;
    
    awsSESSendEmailRequest.message = awsSESMessage;
    AWSSES *awsSES = [AWSSES defaultSES];
    
    [[awsSES sendEmail:awsSESSendEmailRequest] continueWithBlock:^id(AWSTask *task)
     {
         dispatch_async(dispatch_get_main_queue(), ^{
             if (task.error)
             {
                 NSLog(@"awsSES.sendEmail error: %@", task.error.localizedDescription);
             }
             completionHandler(task.error);
         });
         return nil;
     }];
}
```

**Comment by yosuke-matsuda** [comment]
I was able to send an email message using your code snippet. According to [AWS Service Health Dashboard](http://status.aws.amazon.com/), there was an elevated SendEmail, SendRawEmail and SendSMTPEmail API latency. You should try again when the service is normally operating.

**Comment by jeffzacharias** [comment]
Thanks for replying. It wasn't a service issue because the previous version of my app continued to send SES emails fine while the updated version did not. I looked at this for a while and I found:

- If I make a call to the above method to send an SES email immediately after initializing AWS using Cognito, then that SES email gets sent fine, and all other SES emails get sent fine in my app. So, is seems as though at some point in my code after initialing AWS something is executed that makes it so I can no longer send SES email.

- I found that if remove the line of code below to initialize AWSMobileAnalytics then my app can send SES emails fine.
```
mobileAnalytics = [AWSMobileAnalytics mobileAnalyticsForAppId:SharedSettings.mobileAnalyticsAccountID];
```
and then I found that if I change the above line to:
```
mobileAnalytics = [AWSMobileAnalytics mobileAnalyticsForAppId:SharedSettings.mobileAnalyticsAccountID identityPoolId: cognitoId];
```
then the SES emails work fine.

Whenever an SES email does not get sent, AWS responds that everything worked and even with verbose logging turned on there are no errors. And nothing crashes.

Is there any correlation between SES and MobileAnalytics? Does adding the Cognito ID to the MobileAnalytics call affect anything?

**Comment by minimalistcode** [comment]
Thanks for replying. It wasn't a service issue because the previous version of my app continued to send SES emails fine while the updated version did not. I looked at this for a while and I found:

If I make a call to the above method to send an SES email immediately after initializing AWS using Cognito, then that SES email gets sent fine, and all other SES emails get sent fine in my app. So, is seems as though at some point in my code after initialing AWS something is executed that makes it so I can no longer send SES email.

I found that if remove the line of code below to initialize AWSMobileAnalytics then my app can send SES emails fine.
```
mobileAnalytics = [AWSMobileAnalytics mobileAnalyticsForAppId:SharedSettings.mobileAnalyticsAccountID];
```
and then I found that if I change the above line to:
```
mobileAnalytics = [AWSMobileAnalytics mobileAnalyticsForAppId:SharedSettings.mobileAnalyticsAccountID identityPoolId: cognitoId];
then the SES emails work fine.
```
Whenever an SES email does not get sent, AWS responds that everything worked and even with verbose logging turned on there are no errors. And nothing crashes.

Is there any correlation between SES and MobileAnalytics? Does adding the Cognito ID to the MobileAnalytics call affect anything?

**Comment by yosuke-matsuda** [comment]
There is an issue with the `AWSMobileAnalytics`'s use of the default service configuration. We'll address the issue in the next release.

Meanwhile, if you are using CocoaPods, you can modify the SDK code as follows:

`AWSService.h` around line 44
```
/**
 The default service configuration object. This property can be set only once, and any subsequent setters are ignored.
 */
@property (nonatomic, copy) AWSServiceConfiguration *defaultServiceConfiguration;
```

`AWSMobileAnalyticsConfiguration.m` around line 50
```
- (instancetype)init {
    if (self = [super init]) {
        _attributes = [NSDictionary dictionary];
        _useHttps = YES;
        _environment = [AWSMobileAnalyticsEnvironment new];
        _transmitOnWAN = YES;
        _enableEvents = YES;
        _identificationStrategy = AWSAppIdentificationStrategyPrivate;
        _serviceConfiguration = [[AWSServiceManager defaultServiceManager].defaultServiceConfiguration copy];
    }
    return self;
}
```

**Comment by minimalistcode** [comment]
Thanks

**Comment by yosuke-matsuda** [comment]
We have released the AWS Mobile SDK for iOS 2.2.4. It should address your issue. Please let us know if you continue to experience it. Thanks.

**Comment by minimalistcode** [comment]
I tried version 2.2.4 and everything is working fine. Thanks for fixing this.
